### DISCUSSION TITLE
Potential stopwatch tutorial bug or misunderstanding?

### DISCUSSION THREAD

**Discussion by noahgorstein** [Textualize/textual#2407] [Q&A] [answered]
Hey 👋 ! Just want to start by thanking you folks for an awesome project. 

I was just stepping through the stopwatch tutorial. I'm trying to figure out if the following behavior is intended or I'm misunderstanding the docs/tutorial. 

- I start the timer - the timer elapses for ~4s
- Pause the timer - timer now shows ~8s

Demo below from running:

 https://github.com/Textualize/textual/blob/main/docs/examples/tutorial/stopwatch.py

https://user-images.githubusercontent.com/23270779/234941419-e4b87739-14b0-4df0-ba2e-704450f6ad59.mov

I should note I didn't quite follow the `total` attribute added in this step: https://textual.textualize.io/tutorial/#wiring-buttons. Wondering if that's what's causing this seemingly "random" display for pausing/resuming. 

Apologies for the potentially very trivial question but any help much appreciated !

**Comment by TomJGooding** [comment]
Sorry again for misreading your post - just ignore my last reply!

I'm seeing the same behaviour in `stopwatch.py` from v0.21.0, so think this might be a bug rather than any misunderstanding.

**Reply by TomJGooding** [reply]
It looks like there was a change to the `timer` in v0.21.0 which moved a try block - reverting this seems to resolve this issue.

However I confess I don't really understand what this is doing, so I will leave to the maintainers to fix!

https://github.com/Textualize/textual/compare/main...TomJGooding:textual:fix-timer-discussion-2407

**Reply by TomJGooding** [reply, marked as answer]
Fixed by https://github.com/Textualize/textual/pull/2417 in [v0.22.1](https://github.com/Textualize/textual/releases/tag/v0.22.1).

Good spot @noahgorstein!
